### DISCUSSION TITLE
~/.hammerspoon as github repo as external resource

### DISCUSSION THREAD

**Discussion by muescha** [twpayne/chezmoi#2568] [Q&A]
## current situation

- i have my hammerspoon setup in this location `~/.hammerspoon`
- i like to have a separate github repo for the `.hammerspoon`
- i like to use chezmoi
- i am on a single macbook - no other computers

## how i setup it with chezmoi

- i think i need to add it as `github-repo` like described in [External: Include a subdirectory from a git repository](https://www.chezmoi.io/user-guide/include-files-from-elsewhere/#include-a-subdirectory-from-a-git-repository)
- but did this not overwrite my current local repo of `.hammerspoon`?
- how it works well with chezmoi?
- or i did not need this external connection and just have this folder as a repo?

**Comment by twpayne** [comment]
`git-repo` externals cause chezmoi to run `git clone` / `git pull` in the target directory, nothing else. Note that it's `git-repo` and not `github-repo` (i.e. no `hub`).

To debug this further, please post the contents of your `.chezmoiexternal` file and the contents of your `.hammerspoon` repo.

**Reply by muescha** [reply]
i am just in a phase to plan the setup and what is the best choice

i did not created the `.chezmoiexternal` it still now
